### ISSUE TITLE
this with object destructuring

### ISSUE THREAD

**Issue by Ayowel** [eslint/eslint#10356]
**Tell us about your environment**

* **linter-ESLint Version:** 8.4.1
* **Node Version:** 8.9.4
* **npm Version:** 5.6.0

**Configuration:**

object destructuring enabled

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

I have a getter that I don't want to call more than once in a method of the class, hence (reproduced on eslint.org/demo):

```js
class Test {
	get myattr() {
		return 3;
	}
	test() {
		const myattr = { this };
		return myattr;
	}
}
export default Test;
```

**What did you expect to happen?**

Test#test should return 3

**What actually happened? Please include the actual, raw output from ESLint.**

`9:20 - Parsing error: Unexpected keyword 'this'`

**Comment by platinumazure** [comment]
Not sure I understand what you're trying to do. Wouldn't you want to do something like this?

```js
const { myattr } = this;
return myattr;
```
